Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
+	void MiscCmds::Resetrep_Time_limits_load()
+	{
+		ShipnamesWhichAlreadyUsedResetRep.clear();
+		INI_Reader ini;
+		string factionpropfile = resetrepfileadress;
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Playerships"))
+				{
+					uint ids_name;
+					wstring nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
+						{
+							ids_name = ini.get_value_int(0);
+							ShipnamesWhichAlreadyUsedResetRep[nickname] = ids_name;
+						}
+					}
+					
+				}
+			}
+			ini.close();
+			ConPrint(L"Rep: Loaded %u factions\n", factions.size());
+		}
+	}
+
+	void MiscCmds::Resetrep_Time_limits_save()
+	{
+		string factionpropfile = resetrepfileadress;
+
+		FILE *file = fopen(factionpropfile.c_str(), "w");
+		if (file)
+		{
+			fprintf_s(file, "[Playerships]\n");
+			for (map<wstring, uint>::iterator i = ShipnamesWhichAlreadyUsedResetRep.begin();
+				i != ShipnamesWhichAlreadyUsedResetRep.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
+				{
+					wstring temp = i->first;
+					ini_write_wstring(file, "shipname", temp);
+					fprintf_s(file, "timelimit = %d\n", i->second);
+				}
+			}
+			fclose(file);
+		}
+	}
+
+	/** A command to reset all reputations to zero that you might have */
+	bool MiscCmds::UserCmd_ResetRep(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+	{
+		HK_ERROR err;
+
+		wstring wscCharname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+
+		uint iShip;
+		pub::Player::GetShip(iClientID, iShip);
+		if (iShip)
+		{
+			PrintUserCmdText(iClientID, L"ERR Not in base");
+			return true;
+		}
+
+		// Read the current number of credits for the player
+		// and check that the character has enough cash.
+		int iCash = 0;
+		if ((err = HkGetCash(wscCharname, iCash)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR %s", HkErrGetText(err).c_str());
+			return true;
+		}
+
+		if (set_iResetdropCost > 0 && iCash < set_iResetdropCost)
+		{
+			PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+			return true;
+		}
+
+
+		wstring wscCharFileName;
+		if ((err = HkGetCharFileName(wscCharname, wscCharFileName)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR " + HkErrGetText(err));
+			return true;
+		}
+
+		if (ShipnamesWhichAlreadyUsedResetRep.find(wscCharFileName) != ShipnamesWhichAlreadyUsedResetRep.end())
+		{
+			uint LastTime = ShipnamesWhichAlreadyUsedResetRep[wscCharFileName];
+
+			if (((int)time(0) - LastTime) < set_iResetrepTimeLimit)
+			{
+				int SecondsLeft = (set_iResetrepTimeLimit - ((int)time(0) - int(LastTime)));
+				int DaysLeft = SecondsLeft / (3600 * 24);
+			PrintUserCmdText(iClientID, L"ERR Resetrep time limit. %d days (%d in seconds) time remains to wait.", DaysLeft, SecondsLeft);
```

Comment:
broken indentation. also this could be worded better

Reply:
fixed indentation. no idea how to reword for now.

Reply:
`PrintUserCmdText(iClientID, L"ERR Attempt to resetrep too frequently. Please wait %d days (%d in seconds).", DaysLeft, SecondsLeft);`

Reply:
fixed